scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobState.java

```diff
@@ -193,6 +194,104 @@ public JobPriority getPriority() {
      */
     public abstract Map<TaskId, TaskState> getHMTasks();
 
+
+    /**
+     * To get the task as an array list and filtered by a given tag.
+     * @param tag the used to filter the tasks.
+     * @return the set of filtered task states.
+     */
+    public List<TaskState> getTaskByTag(final String tag){
+        List<TaskState> tasks = this.getTasks();
+        return (List<TaskState>) CollectionUtils.select(tasks, new Predicate() {
```

Comment:
We are already doing the same logic in the JobStateImpl. We need to avoid duplications.